### DISCUSSION TITLE
Keymap Not Working As Expected

### DISCUSSION THREAD

**Discussion by josephcrawfordSRH** [neovim/neovim#30947] [General]
I have the following keymap defined

```lua
vim.keymap.set("n", "<leader>dd", "ggdG", { desc = "Delete entire document" })
 ```

I would expect that it would delete the entire document content.  If I am on the first line of the document it will work. 

If I am on say line 300 and I execute this keymap it then sometimes it works but 90% of the time it simply jumps me to the first line and stops.  Am I doing something wrong here?  Is there something with the way keymaps work that I am misunderstanding?

**Comment by SeanSelene** [comment]
It always works for me. Maybe it influenced by other keymaps. By the way, LazyVim added a textobject "g" which refers to the whole buffer, so you can use "dig" if you are using Lazyvim. Or you can add it yourself.

**Reply by josephcrawfordSRH** [reply]
thanks for that information, I was not aware of that!

**Comment by attilarepka** [comment]
Try using [neovim's internal API](https://neovim.io/doc/user/api.html#nvim_buf_set_lines()):

``` lua
vim.keymap.set("n", "<leader>dd", function()
    vim.api.nvim_buf_set_lines(0, 0, -1, false, {})
end, { desc = "Delete entire document" })
```

**Reply by josephcrawfordSRH** [reply]
@attilarepka this appears to work perfectly, thank you!  I need to learn more about these api's!
